CsharpExtras/Dictionary/MultiValueMapImpl.cs

```diff
@@ -168,5 +154,25 @@ public override bool Equals(object obj)
             }
             return true;
         }
+
+        public override int GetHashCode()

```

Comment:
The issue with this is it becomes quite inefficient. We have a hashcode generation function that's linear in the number of elements in all the value sets. Considering the hash code is used to store elements in things like hash sets, which are supposed to be efficient, this is problematic.

Perhaps a better solution is to remove the ``equals`` override and instead add another equality method e.g. ``mapEquals`` that's analogoug to the C# function ``setEquals`` that's in the set interface.

Reply:
I think getting a hashcode is always going to be inefficient. If you are calculating a hash for any data type (string, list, set, etc) you have to iterate over the whole thing at least once to calculate the hash. With equality you can sometimes break early, but not when calculating a hash.

There is another issue with the hashcode though that we should consider: [stackoverflow question](https://stackoverflow.com/questions/10567450/implement-gethashcode-for-objects-that-contain-collections). The idea is that if you setup your hashcode definition to change when the data in the object changes, and if your object happens to be in a set or dictionary, it can get lost.